tests/v2/validation/provisioning/README.md

```diff
@@ -0,0 +1,294 @@
+
+## RKE2 Provisioning Configs
+
+---
+
+### Provisioning Input
+provisioningInput is needed to the run the RKE2 tests, specifically kubernetesVersion, cni, and providers. nodesAndRoles is only needed for the TestProvisioningDynamicInput test, nodes are divided by "|". nodeProviders is only needed for custom cluster test; the framework only supports custom clusters through aws/ec2 instances.
+
+```json
+"provisioningInput": {
+    "nodesAndRoles": "etcd,controlplane,worker|etcd,controlplane",
+    "kubernetesVersion": ["v1.21.6+rke2r1"],
+    "cni": ["calico"],
+    "providers": ["linode", "aws", "do", "harvester"],
+    "nodeProviders": ["ec2"]
+  }
+```
+
+### AWS/EC2 Config
+For custom clusters the below config is needed, only AWS/EC2 
+
+```json
+ "awsEC2Config": {
+    "region": "us-east-2",
+    "instanceType": "t3a.medium",
+    "awsRegionAZ": "",
+    "awsAMI": "",
+    "awsSecurityGroups": [""],
+    "awsAccessKeyID": "",
+    "awsSecretAccessKey": "",
+    "awsSSHKeyName": "",
+    "awsCICDInstanceTag": "",
+    "awsIAMProfile": "",
+    "awsUser": "ubuntu",
+    "volumeSize": 50
+  },
+```
+
+### Cloud Credentials
+These are the inputs needed for the different node provider cloud credentials, inlcuding linode, aws, digital ocean, harvester, azure, and google.
+
+```json
+"digitalOceanCredentials": {
+   "accessToken": ""
+  },
+"linodeCredentials": {
+   "token": ""
+  },
+"azureCredentials": {
+   "clientId": "",
+   "clientSecret": "",
+     "subscriptionId": "",
+     "environment": "AzurePublicCloud"
+  },
+"awsCredentials": {
+   "secretKey": "",
+   "accessKey": "",
+   "defaultRegion": ""
+  },
+"harvesterCredentials": {
+   "clusterId": "",
+   "clusterType": "",
+   "kubeconfigContent": ""
+},
+"googleCredentials": {
+    "authEncodedJson": ""
+}
+```
+### Machine RKE Config
+Machine RKE pool config is the last thing need for the RKE2 provisioning tests
+
+```json
+"awsMachineConfig": {
+    "region": "us-east-2",
+    "instanceType": "t3a.medium",
+    "sshUser": "ubuntu",
+    "vpcId": "",
+    "volumeType": "gp2",
+    "zone": "a",
+    "retries": "5",
+    "rootSize": "16",
+    "securityGroup": ["rancher-nodes"]
+  },
+  "doMachineConfig": {
+    "image": "ubuntu-20-04-x64",
+      "backups": false,
+      "ipv6": false,
+      "monitoring": false,
+      "privateNetworking": false,
+      "region": "nyc3",
+      "size": "s-2vcpu-4gb",
+      "sshKeyContents": "",
+      "sshKeyFingerprint": "",
+      "sshPort": "22",
+      "sshUser": "root",
+      "tags": "",
+      "userdata": ""
+  },
+  "linodeMachineConfig": {
+    "authorizedUsers": "",
+    "createPrivateIp": false,
+    "dockerPort": "2376",
+    "image": "linode/ubuntu20.04",
+    "instanceType": "g6-standard-2",
+    "region": "us-west",
+    "rootPass": "",
+    "sshPort": "22",
+    "sshUser": "",
+    "stackscript": "",
+    "stackscriptData": "",
+    "swapSize": "512",
+    "tags": "",
+    "uaPrefix": ""
+  },
+  "azureMachineConfig": {
```

Comment:
Could we split these big json blocks to their respective cloud provider sub title?

Like: `## Azure example ##` then the json and the next title `## EKS example ##` (or a deeper subtitle level).

I've seen the doc in view document and that would help its readability

Reply:
Does that look better @izaac ?

Reply:
Way better! thank you!